FreeRTOS/Test/Target/tests/fr2/test.c

```diff
@@ -0,0 +1,164 @@
+as ( configMAX_PRIORITIES - 1 ). */
+#define mainTASK_A_PRIORITY (tskIDLE_PRIORITY + 2)
+#define mainTASK_B_PRIORITY (tskIDLE_PRIORITY + 1)
+#define mainTASK_C_PRIORITY (tskIDLE_PRIORITY + 3)
+
+static void prvTaskA(void *pvParameters);
+static void prvTaskB(void *pvParameters);
+static void prvTaskC(void *pvParameters);
+
+#if configNUMBER_OF_CORES != 2
+#error Require two cores be configured for FreeRTOS
+#endif
+
+static bool testPassed = false;
+static bool testFailed = false;
+
+void test_fr2TASK_SWITCHED_IN(void) {
+  UBaseType_t idx, numTasksRunning;
+  TaskStatus_t taskStatus[16];
+  UBaseType_t taskStatusArraySize = 16;
+  unsigned long totalRunTime;
+  int coreIndex = 0;
+  SchedTraceLogRow *logRow;
+  int retcode = 0;
+
+  static int taskSwitchCount = 0;
+  static bool taskARan = false;
+  static bool taskBRan = false;
+  static bool taskCRan = false;
+
+  if (!(testPassed || testFailed))
+  {
+    numTasksRunning = uxTaskGetSystemState((TaskStatus_t * const)&taskStatus, taskStatusArraySize, &totalRunTime);
+
+    for(idx = 0; idx < numTasksRunning; idx++)
+    {
+      if ((strcmp(taskStatus[idx].pcTaskName, "TaskA") == 0) && (taskStatus[idx].eCurrentState == eRunning))
+      {
+        taskARan = true;
+      }
+      if ((strcmp(taskStatus[idx].pcTaskName, "TaskB") == 0) && (taskStatus[idx].eCurrentState == eRunning))
+      {
+        taskBRan = true;
+      }
+      if ((strcmp(taskStatus[idx].pcTaskName, "TaskC") == 0) && (taskStatus[idx].eCurrentState == eRunning))
+      {
+        taskCRan = true;
+      }
+    }
+
+    if (taskBRan)
+    {
+      if (!(taskARan && taskCRan))
+      {
+        testFailed = true;
+      }
+      else
+      {
+        testPassed = true;
+      }
+    }
+
+    taskSwitchCount++;
+    if (taskSwitchCount > 2048)
+    {
+      testFailed = true;
+    }
+  }
+}
+
+void setup_test_fr2_001(void) {
+  xTaskCreate(prvTaskA, "TaskA", configMINIMAL_STACK_SIZE, NULL,
+              mainTASK_A_PRIORITY, NULL);
+
+  xTaskCreate(prvTaskB, "TaskB", configMINIMAL_STACK_SIZE, NULL,
+              mainTASK_B_PRIORITY, NULL);
+
+  xTaskCreate(prvTaskC, "TaskC", configMINIMAL_STACK_SIZE, NULL,
+              mainTASK_C_PRIORITY, NULL);
+}
+
+void setUp(void) {} /* Is run before every test, put unit init calls here. */
+void tearDown(void) {
+} /* Is run after every test, put unit clean-up calls here. */
+
+int main(void) {
+  initTestEnvironment();
+
+  setup_test_fr2_001();
+
+  vTaskStartScheduler();
+
+  /* should never reach here */
+  panic_unsupported();
+
+  return 0; // UNITY_END is unreachable via this path. a state machine and
+            // counter is used so that just one child task will call it
+            // instead.
+}
+
+static void prvTaskA(void *pvParameters) {
+  // idle the task
+  for (;;) {
+    vTaskDelay(pdMS_TO_TICKS(10));
+    busyWaitMicroseconds(100000);
+  }
+}
+
+static void fr02_validateHigherPriorityTasksAlreadyRan(void)
```

Comment:
FR-2's first summary is 'he scheduler shall correctly schedule the highest priority ready tasks.'

To me this means it will schedule higher priority tasks before lower priority tasks.